File: src/MyStack.java
public class MyStack <T> {
    private final int maxSize;
    private final T[] data;
    private int top;

    @SuppressWarnings("unchecked")
    public MyStack(int size) {
        maxSize = size;
        data = (T[]) new Object[size];
        top = -1;
    }

    public void push(T val) {
        data[++top] = val;
    }

    public T pop() {
        T temp = data[top];
        data[top--] = null;
        return temp;
    }

    public T peek() {
        return data[top];
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == maxSize - 1;
    }

    public int size() {
        return top + 1;
    }
}

File: Calculator/app/src/main/java/com/example/calculator/MyStack.kt
package com.example.calculator

class MyStack<T>(private val maxSize: Int) {
    private val data = arrayOfNulls<Any>(maxSize) as Array<T?>
    private var top: Int = -1

    fun push(`val`: T) {
        data[++top] = `val`
    }

    fun pop(): T? {
        val temp = data[top]
        data[top--] = null
        return temp
    }

    fun peek(): T? {
        return data[top]
    }

    fun isEmpty(): Boolean {
        return top == -1
    }

    fun isFull(): Boolean {
        return top == maxSize - 1
    }

    fun size(): Int {
        return top + 1
    }

}


